src/validation/signupValidation.js

```diff
@@ -0,0 +1,16 @@
+import Check from '../utils/validator';
+import sendResult from '../utils/sendResult';
+
+const validateSignup = (req, res, next) => {
+  const valid = [
+    new Check({ email: req }).req().str().email()
+      .min(4),
+    new Check({ username: req }).req().str().min(2),
```

Comment:
users password should contain any special character like %^( so consider specifying either numeric or alphaNumeric